### ISSUE TITLE
Crash on apply the second time

### ISSUE THREAD

**Issue by mill5james** [hashicorp/terraform#28370]
I have a simple plan that worked before updating to v0.15.0. 

``` shell
> terraform -version                                         
Terraform v0.15.0
on windows_amd64
+ provider registry.terraform.io/hashicorp/helm v2.1.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.0.3
```

The plan deploys to a local kubernetes running in Docker for Windows PostgreSQL and Airflow. There is nothing special about it. Here is a [public gist](https://gist.github.com/mill5james/261e7863a54c05df8a1543a52e32a774) to the code.

This will work the first time, when the environment does not exist.
When running the second time (even when the Terraform plan is unmodified)  I get the following crash:

```
panic: value is marked, so must be unmarked first

goroutine 359 [running]:
github.com/zclconf/go-cty/cty.Value.assertUnmarked(...)
        /go/pkg/mod/github.com/zclconf/go-cty@v1.8.1/cty/marks.go:123
github.com/zclconf/go-cty/cty.Value.LengthInt(0x308a028, 0xc000bf8df0, 0x29342e0, 0xc0006cb0c8, 0x308a001)
        /go/pkg/mod/github.com/zclconf/go-cty@v1.8.1/cty/value_ops.go:1045 +0x56
github.com/zclconf/go-cty/cty.transform(0xc000bf93a0, 0x1, 0x1, 0x308a028, 0xc000bf8df0, 0x29342e0, 0xc0006cb0c8, 0x3053950, 0xc0006cb128, 0xc0006cb0c8, ...)
        /go/pkg/mod/github.com/zclconf/go-cty@v1.8.1/cty/walk.go:144 +0x2f9
github.com/zclconf/go-cty/cty.transform(0x0, 0x0, 0x0, 0x3089ff0, 0xc000bf9300, 0x2845800, 0xc000bed4a0, 0x3053950, 0xc0006cb128, 0x0, ...)
        /go/pkg/mod/github.com/zclconf/go-cty@v1.8.1/cty/walk.go:209 +0x10ed
github.com/zclconf/go-cty/cty.TransformWithTransformer(...)
        /go/pkg/mod/github.com/zclconf/go-cty@v1.8.1/cty/walk.go:125
github.com/zclconf/go-cty/cty.Value.MarkWithPaths(0x3089ff0, 0xc000bf9300, 0x2845800, 0xc000bed4a0, 0xc000d86f80, 0x1, 0x1, 0x0, 0xc000d86f80, 0x1, ...)
        /go/pkg/mod/github.com/zclconf/go-cty@v1.8.1/cty/marks.go:215 +0xcb
github.com/hashicorp/terraform/terraform.markProviderSensitiveAttributes(0xc000a4ba40, 0x3089ff0, 0xc000bf9300, 0x2845800, 0xc000bed4a0, 0x0, 0xc, 0xc00016d0f0, 0xa)
        /home/circleci/project/project/terraform/evaluate.go:960 +0xc5
github.com/hashicorp/terraform/terraform.(*evaluationStateData).GetResource(0xc0018a27e0, 0x4d, 0xc00016d0e0, 0xc, 0xc00016d0f0, 0xa, 0xc00016c8a9, 0x7, 0x88, 0x10, ...)
        /home/circleci/project/project/terraform/evaluate.go:787 +0x1353
github.com/hashicorp/terraform/lang.(*Scope).evalContext(0xc00100f180, 0xc0003ab300, 0x8, 0x8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x30)
        /home/circleci/project/project/lang/eval.go:360 +0x208d
github.com/hashicorp/terraform/lang.(*Scope).EvalContext(...)
        /home/circleci/project/project/lang/eval.go:238
github.com/hashicorp/terraform/lang.(*Scope).EvalBlock(0xc00100f180, 0x3088ab8, 0xc001797d10, 0xc000a4ba40, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
        /home/circleci/project/project/lang/eval.go:51 +0x105
github.com/hashicorp/terraform/terraform.(*BuiltinEvalContext).EvaluateBlock(0xc00075b930, 0x3088d58, 0xc001797d10, 0xc000a4ba40, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/circleci/project/project/terraform/eval_context_builtin.go:273 +0x1bb
github.com/hashicorp/terraform/terraform.(*NodeAbstractResourceInstance).plan(0xc0018a61c0, 0x30c0990, 0xc00075b930, 0xc0006b3180, 0xc0014a8f60, 0x0, 0x0, 0x0, 0x0, 0xc00016cf77, ...)
        /home/circleci/project/project/terraform/node_resource_abstract_instance.go:577 +0x3df
github.com/hashicorp/terraform/terraform.(*NodeApplyableResourceInstance).managedResourceExecute(0xc000c102a0, 0x30c0990, 0xc00075b930, 0x0, 0x0, 0xc00062dc10)
        /home/circleci/project/project/terraform/node_resource_apply_instance.go:235 +0x63d
github.com/hashicorp/terraform/terraform.(*NodeApplyableResourceInstance).Execute(0xc000c102a0, 0x30c0990, 0xc00075b930, 0xc00009a001, 0xc00062dd18, 0x7ec06a, 0x288f5c0)
        /home/circleci/project/project/terraform/node_resource_apply_instance.go:129 +0xbc
github.com/hashicorp/terraform/terraform.(*ContextGraphWalker).Execute(0xc0008a7380, 0x30c0990, 0xc00075b930, 0x2326381e9b0, 0xc000c102a0, 0x0, 0x0, 0x0)
        /home/circleci/project/project/terraform/graph_walk_context.go:127 +0xc6
github.com/hashicorp/terraform/terraform.(*Graph).walk.func1(0x2bdb460, 0xc000c102a0, 0x0, 0x0, 0x0)
        /home/circleci/project/project/terraform/graph.go:59 +0xbb3
github.com/hashicorp/terraform/dag.(*Walker).walkVertex(0xc00048a540, 0x2bdb460, 0xc000c102a0, 0xc0010089c0)
        /home/circleci/project/project/dag/walk.go:381 +0x2a2
created by github.com/hashicorp/terraform/dag.(*Walker).Update
        /home/circleci/project/project/dag/walk.go:304 +0x1246




!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.471382662.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

SECURITY WARNING: the "crash.471382662.log" file that was created may contain
sensitive information that must be redacted before it is safe to share
on the issue tracker.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```
[crash.577544246.log](https://github.com/hashicorp/terraform/files/6313533/crash.577544246.log)

**Comment by alisdair** [comment]
@mill5james Thanks for reporting this! I'm investigating it now and I have an idea of what's wrong.

I'm having some trouble applying your given configuration. After installing minikube and starting it, running `terraform apply` fails with the following error:

```
│ Error: Post "http://localhost/api/v1/namespaces": dial tcp [::1]:80: connect: connection refused
│
│   with kubernetes_namespace.default,
│   on main.tf line 49, in resource "kubernetes_namespace" "default":
│   49: resource "kubernetes_namespace" "default" {
│
```

Am I missing something obvious about how to apply this configuration?

**Comment by alisdair** [comment]
Based on the stack trace, I believe this should be fixed by #28381.

**Comment by mill5james** [comment]
There is really nothing special about this. The design is to support an "out of the box" experience for developer needing to develop for Airflow who have limited knowledge of Docker, Kubernetes, and Terraform. The concept was "if you can install Docker Desktop for Windows and Terraform, you can stand up a dev environment in very few command"

I was not using minikube but the [Docker Desktop Kubernetes](https://docs.docker.com/desktop/kubernetes/).
I did just install minikube, changed the `kube_context` variable to `minikube` and can reproduce the problem in minikube

**Comment by wgebis** [comment]
@jbardin do you plan release a hotfix for that bug?

**Comment by mill5james** [comment]
@alisdair 
> I'm having some trouble applying your given configuration. After installing minikube and starting it, running `terraform apply` fails with the following error:
> 
> ```
> │ Error: Post "http://localhost/api/v1/namespaces": dial tcp [::1]:80: connect: connection refused
> │
> │   with kubernetes_namespace.default,
> │   on main.tf line 49, in resource "kubernetes_namespace" "default":
> │   49: resource "kubernetes_namespace" "default" {
> │
> ```
> 
> Am I missing something obvious about how to apply this configuration?

Ok. I can reproduce your issue. The problem is the `kube_context` being set incorrectly for minikube. 

The default value in my `variables.tf` is `docker-desktop`. For minikube, set `kube_context` = `minikube`.
This error comes from the kubenetes provider when attempting to connect to the wrong context.

**Comment by alisdair** [comment]
Thanks very much, @mill5james! Not sure how I missed that, but with that in place I was able to reproduce the issue and also confirm that it is now fixed on the main branch.

@wgebis:

> do you plan release a hotfix for that bug?

The fix will be released in 0.15.1, which will either be next Wednesday or the following week.
